bot/exts/moderation/clean.py

```diff
@@ -0,0 +1,580 @@
+
+        logged = await self._log_clean(deleted_messages, channels, ctx)
+
+        if logged and is_mod_channel(ctx.channel):
+            with suppress(NotFound):  # Can happen if the invoker deleted their own messages.
+                await ctx.message.add_reaction(Emojis.check_mark)
+
+    # region: Commands
+
+    @group(invoke_without_command=True, name="clean", aliases=["clear", "purge"])
+    async def clean_group(
+            self,
+            ctx: Context,
+            traverse: Optional[int] = None,
+            users: Greedy[User] = None,
+            first_limit: Optional[CleanLimit] = None,
+            second_limit: Optional[CleanLimit] = None,
+            use_cache: Optional[bool] = None,
+            bots_only: Optional[bool] = False,
+            regex: Optional[Regex] = None,
+            *,
+            channels: CleanChannels = None
+    ) -> None:
+        """
+        Commands for cleaning messages in channels.
+
+        If arguments are provided, will act as a master command from which all subcommands can be derived.
+        `traverse`: The number of messages to look at in each channel.
+        `users`: A series of user mentions, ID's, or names.
+        `first_limit` and `second_limit`: A message, a duration delta, or an ISO datetime.
+        If a message is provided, cleaning will happen in that channel, and channels cannot be provided.
+        If a limit is provided, multiple channels cannot be provided.
+        If only one of them is provided, acts as `clean until`. If both are provided, acts as `clean between`.
+        `use_cache`: Whether to use the message cache.
+        If not provided, will default to False unless an asterisk is used for the channels.
+        `bots_only`: Whether to delete only bots. If specified, users cannot be specified.
+        `regex`: A regex pattern the message must contain to be deleted.
+        The pattern must be provided enclosed in backticks.
+        If the pattern contains spaces, it still needs to be enclosed in double quotes on top of that.
+        `channels`: A series of channels to delete in, or an asterisk to delete from all channels.
+        """
+        if not any([traverse, users, first_limit, second_limit, regex, channels]):
+            await ctx.send_help(ctx.command)
+            return
+
+        if not traverse:
+            if first_limit:
+                traverse = CleanMessages.message_limit
+            else:
+                traverse = DEFAULT_TRAVERSE
+        if not use_cache:
+            use_cache = channels == "*"
+
+        await self._clean_messages(
+            ctx, traverse, channels, bots_only, users, regex, first_limit, second_limit, use_cache
+        )
+
+    @clean_group.command(name="user", aliases=["users"])
+    async def clean_user(
+        self,
+        ctx: Context,
+        user: User,
+        traverse: Optional[int] = 10,
+        use_cache: Optional[bool] = True,
+        *,
+        channels: CleanChannels = None
```

Comment:
Could we add a comment here, for why we don't use `Optional[CleanChannels]` here? I didn't understand the need for it until I saw [`3ccb533` (#1793)](https://github.com/python-discord/bot/pull/1793/commits/3ccb533686d464e11bf330ac19900a9f6cfc4366).